### PR TITLE
chore: remove appium module from npm-shrinkwrap.json

### PR DESCRIPTION
**Pull request by KazuCocoa**
https://github.com/appium/appium/discussions/17112#discussioncomment-3057952

Let's see how this will help.
On my local, actually removing `node_modules/appium` helped to remove it from `npm-shrinkwrap.json`

### CODE REVIEW

File: .github/workflows/publish.js.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
     - run: npm install
       name: Install dependencies
     - run: rm -rf package-lock.json
+    - run: rm -rf node_modules/appium
```
Comment (mykola-mokhnach) [inline .github/workflows/publish.js.yml:24]: I'm not quite sure we should do this before running tests as they would most likely fail.
We should also probably be smart here and remove all dependencies that are listed as peer ones in package.json

Should the same change be applied to other drivers btw?

File: .github/workflows/publish.js.yml [hunk 2]
```diff
@@ -21,6 +21,7 @@ jobs:
...
     - run: npm shrinkwrap
```
Comment (mykola-mokhnach) [inline .github/workflows/publish.js.yml:25]: do we shrinkwrap dev dependencies as well?

Comment (mykola-mokhnach) [reply .github/workflows/publish.js.yml:25]: perhaps we should apply something like https://github.com/npm/npm/issues/11189

Comment (KazuCocoa) [reply .github/workflows/publish.js.yml:25]: Maybe:

1. run npm run test first after `npm install`
2. remove all node_modules and runs `npm install --production`
3. create `shrinkwrap`

can do. One possible concern is modules installed in step2) could be different from 1)'s npm run one (e.g. a new module version has been released between them).

Comment (KazuCocoa) [reply .github/workflows/publish.js.yml]: The current change should be ok.

File: .github/workflows/publish.js.yml [hunk 3]
```diff
@@ -20,11 +20,15 @@ jobs:
...
-    - run: rm -rf package-lock.json
-    - run: npm shrinkwrap
-      name: Create shrinkwrap
     - run: npm test
       name: Run NPM Test
+    - run: |
+        rm -rf package-lock.json
+        npm prune --production
+        rm -rf node_modules/appium
+      name: Remove deps dependencies and appium peerDependencies
```
Comment (mykola-mokhnach) [inline .github/workflows/publish.js.yml:29]: deps->dev

Comment (mykola-mokhnach) [reply .github/workflows/publish.js.yml:29]: peerDependencies -> peer dependencies

**Review by mykola-mokhnach — APPROVED** [review]

**Review by jlipps — COMMENTED** [review]
nice! seems strange that shrinkwrap would include any dev or peer deps.

**Comment by boneskull** [comment]
oh.. you may be able to just do `npm shrinkwrap --production` instead

**Comment by boneskull** [comment]
@KazuCocoa try it?

**Comment by KazuCocoa** [comment]
oh, not yet. Let me try the way later

**Comment by KazuCocoa** [comment]
hm, with/without production generated the same file

**Pull request merged**